mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusLifeCycleTest.java

```diff
@@ -0,0 +1,262 @@
+import static org.apache.james.backend.rabbitmq.Constants.DIRECT_EXCHANGE;
+import static org.apache.james.mailbox.events.RabbitMQEventBus.MAILBOX_EVENT_EXCHANGE_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.apache.james.backend.rabbitmq.RabbitMQConnectionFactory;
+import org.apache.james.backend.rabbitmq.RabbitMQExtension;
+import org.apache.james.backend.rabbitmq.RabbitMQManagementAPI;
+import org.apache.james.event.json.EventSerializer;
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.mailbox.model.TestMessageId;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+class RabbitMQEventBusLifeCycleTest {
+
+    static class RabbitMQEventBusDeclarationExtension implements BeforeEachCallback, AfterEachCallback {
+
+        private EventSerializer eventSerializer;
+        private RabbitMQConnectionFactory connectionFactory;
+        private RabbitMQManagementAPI rabbitMQManagement;
+        private RoutingKeyConverter routingKeyConverter;
+
+        @Override
+        public void beforeEach(ExtensionContext context) throws Exception {
+            rabbitMQExtension.beforeEach(context);
+
+            rabbitMQManagement = rabbitMQExtension.managementAPI();
+            connectionFactory = rabbitMQExtension.getConnectionFactory();
+
+            TestId.Factory mailboxIdFactory = new TestId.Factory();
+            routingKeyConverter = RoutingKeyConverter.forFactories(new MailboxIdRegistrationKey.Factory(mailboxIdFactory));
+            eventSerializer = new EventSerializer(mailboxIdFactory, new TestMessageId.Factory());
+        }
+
+        @Override
+        public void afterEach(ExtensionContext context) throws Exception {
+            rabbitMQExtension.afterEach(context);
+        }
+
+        private RabbitMQEventBus newEventBus() {
+            return new RabbitMQEventBus(connectionFactory, eventSerializer, routingKeyConverter);
+        }
+    }
+
+    private static class GroupA extends Group {}
+
+    static RabbitMQExtension rabbitMQExtension = new RabbitMQExtension();
+    @RegisterExtension
+    static RabbitMQEventBusDeclarationExtension testExtension = new RabbitMQEventBusDeclarationExtension();
+
+    @BeforeAll
+    static void setUp() {
+        rabbitMQExtension.beforeAll(null);
```

Comment:
this trick helps avoiding start 2 rabbitmq docker, because there are 2 nested test classes, JUNIT 5 treat `RegisterExtension` in class level life cycle, then do not to mark @ RegisterExtension to `static RabbitMQExtension rabbitMQExtension = new RabbitMQExtension();`

Reply:
Maybe you should write a comment about it.